src/Storages/Hive/HiveFile.cpp

```diff
@@ -119,65 +124,192 @@ Range HiveOrcFile::buildRange(const orc::ColumnStatistics * col_stats)
 
 void HiveOrcFile::prepareReader()
 {
-    // TODO To be implemented
-    throw Exception("Unimplemented HiveOrcFile::prepareReader", ErrorCodes::NOT_IMPLEMENTED);
+    in = std::make_unique<ReadBufferFromHDFS>(namenode_url, path, getContext()->getGlobalContext()->getConfigRef());
+    auto format_settings = getFormatSettings(getContext());
+    std::atomic<int> is_stopped{0};
+    auto result = arrow::adapters::orc::ORCFileReader::Open(asArrowFile(*in, format_settings, is_stopped), arrow::default_memory_pool());
+    THROW_ARROW_NOT_OK(result.status());
+    reader = std::move(result).ValueOrDie();
 }
 
 void HiveOrcFile::prepareColumnMapping()
 {
-    // TODO To be implemented
-    throw Exception("Unimplemented HiveOrcFile::prepareColumnMapping", ErrorCodes::NOT_IMPLEMENTED);
+    const orc::Type & type = reader->GetRawORCReader()->getType();
+    size_t size = type.getSubtypeCount();
+    for (size_t pos = 0; pos < size; pos++)
+    {
+        /// Column names in hive is case-insensitive.
+        String column{type.getFieldName(pos)};
+        boost::to_lower(column);
+        orc_column_positions[column] = pos;
+    }
 }
 
 bool HiveOrcFile::hasMinMaxIndex() const
 {
-    return false;
+    return !storage_settings->enable_orc_file_minmax_index;
```

Comment:
why `!`? may be should be `has` == `enable`

Reply:
My mistake, fixed.